src/splice.test.ts

```diff
@@ -0,0 +1,29 @@
+import { pipe } from './pipe';
+import { splice } from './splice';
+
+describe('data first', () => {
+  test('splice', () => {
+    expect(splice([1, 2, 3, 4, 5, 6, 7, 8], 2, 3, [])).toEqual([1, 2, 6, 7, 8]);
+    expect(splice([1, 2, 3, 4, 5, 6, 7, 8], 2, 3, [9, 10])).toEqual([
+      1, 2, 9, 10, 6, 7, 8,
+    ]);
```

Comment:
I'd split this into several tests. There's more value in having a unique test for each case (here: with replacement and without).

Reply:
I fixed test case and assertion to be one-to-one. 

> This is great, just a few nits. Can you also add a few more tests to cover all the edge cases (index > length, index < 0, deleteCount > length, deleteCount < 0, etc...) so that if anyone changes the impl, they will know if they are changing an output that someone might be expecting.

I added regression testing. I tested with matrix data to include as many edge cases as possible.

cf. https://github.com/remeda/remeda/pull/393/files#diff-0696fb3eb6df2fdeb8e06f4dab9cd586776e32d60d7001dc8eb02cd00ad58061